cgp/genome.py

```diff
@@ -267,6 +267,181 @@ def randomize(self, rng: np.random.RandomState) -> None:
         # accept generated dna if it is valid
         self.dna = dna
 
+    def reorder(self, rng: np.random.RandomState) -> None:
+        """Reorder the genome
+
+        Shuffle node ordering of internal (hidden) nodes in genome without changing node behavior.
+        (Goldman 2015, DOI: 10.1109/TEVC.2014.2324539)
+
+        From a set of nodes with no dependencies (addable_nodes)
+        a (pseudo-) random node is placed at the first internal position.
+        After updating the set of addable nodes,
+        a (pseudo-) random node for the next position is picked.
+        This procedure is repeated until all nodes are placed.
+
+        Gene values are updated to correspond to the new address of the corresponding node.
+
+        Invalid gene values (from unused arities of eg. constant operator nodes)
+        are replaced by (pseudo-) random picks of permissible values for that position.
```

Comment:
```suggestion
        During reordering, inactive genes, e.g., input genes of nodes with arity zero, are not taken
        into account and can hence have invalid values after reordering.
        These invalid values are replaced by random values for the respective gene after reordering.
```
it's important to note this, but the rest of the explanation can be removed i think